content/courses/onchain-development/intro-to-anchor.md

```diff
@@ -107,43 +130,55 @@ pub struct Context<'a, 'b, 'c, 'info, T> {
     /// Bump seeds found during constraint validation. This is provided as a
     /// convenience so that handlers don't have to recalculate bump seeds or
     /// pass them in as arguments.
-    pub bumps: BTreeMap<String, u8>,
+    /// Type is the bumps struct generated by #[derive(Accounts)]
+    pub bumps: T::Bumps,
 }
 ```
 
 `Context` is a generic type where `T` defines the list of accounts an
 instruction requires. When you use `Context`, you specify the concrete type of
-`T` as a struct that adopts the `Accounts` trait (e.g.
-`Context<AddMovieReviewAccounts>`). Through this context argument the
-instruction can then access:
+`T` as a struct that adopts the `Accounts` trait.
+
+The first argument of every instruction handler must be `Context`. `Context`
+takes a generic of your `Accounts` struct, eg, if `AddMovieReview` was the
+struct holding the accounts, the context for the `add_movie_review()` function
+would be `Context<AddMovieReview>`.
+
+<callout type="info" title="Naming convention">
+  Yes, the Accounts struct is typically named the same thing as the instruction handler, just in TitleCase. Eg, the struct with the accounts for add_movie_review() is called AddMovieReview!
+</callout>
+
+Through this context argument the instruction can then access:
 
 - The accounts passed into the instruction (`ctx.accounts`)
 - The program ID (`ctx.program_id`) of the executing program
 - The remaining accounts (`ctx.remaining_accounts`). The `remaining_accounts` is
   a vector that contains all accounts that were passed into the instruction but
   are not declared in the `Accounts` struct.
 - The bumps for any PDA accounts in the `Accounts` struct (`ctx.bumps`)
+- The seeds for any PDA accounts in tha `Accounts` struct (`ctx.seeds`)
+
+> The design of Contexts can be different across different programs to serve
+> their purpose; and the name of the context could be anything (not limited to
+> Context) to better reflect it's usage. This example is to help understand how
+> contexts work in Anchor.
 
 ### Define instruction accounts
 
-The `Accounts` trait defines a data structure of validated accounts. Structs
-adopting this trait define the list of accounts required for a given
-instruction. These accounts are then exposed through an instruction's `Context`
-so that manual account iteration and deserialization is no longer necessary.
+The `Accounts` trait:
 
-You typically apply the `Accounts` trait through the `derive` macro (e.g.
-`#[derive(Accounts)]`). This implements an `Accounts` deserializer on the given
-struct and removes the need to deserialize each account manually.
+- Defines a structure of validated accounts for an instruction
+- Makes accounts accessible through an instruction's `Context`
```

Comment:
Can you please go through and fix usage of 'instruction handler' (for the function) versus 'instruction' (for the input to the function)? See CONTRIBUTING.md

Reply:
yes sure

Reply:
did it